src/Core/Services/Implementations/PolicyService.cs

```diff
@@ -127,5 +131,59 @@ public class PolicyService : IPolicyService
             await _policyRepository.UpsertAsync(policy);
             await _eventService.LogPolicyEventAsync(policy, Enums.EventType.Policy_Updated);
         }
+
+        public async Task<bool> PolicyAppliesToCurrentUserAsync(PolicyType policyType,
```

Comment:
Generally I'm not a fan of using CurrentContext in the service. I know sometimes it's necessary, but I don't think it is right now. This method has more value for reuse later if we just pass UserId in as a parameter, and determine what user that is based on what API endpoint is hit (pass CurrentContext.UserId.Value from the controller).